ansible_bender/callback_plugins/snapshoter.py

```diff
@@ -69,17 +72,50 @@ def _snapshot(self, task_result):
 
     @staticmethod
     def get_task_content(task: Task):
+        sha512 = hashlib.sha512()
         serialized_data = task.get_ds()
         if not serialized_data:
             # ansible 2.8
             serialized_data = task.dump_attrs()
         if not serialized_data:
             logger.error("unable to obtain task content from ansible: caching will not work")
             return
-        c = json.dumps(serialized_data, sort_keys=True).encode("utf-8")
+        c = json.dumps(serialized_data, sort_keys=True)
+
         logger.debug("content = %s", c)
-        m = hashlib.sha512(c)
-        return m.hexdigest()
+        sha512.update(c.encode("utf-8"))
+
+        # If task is a file action, cache the src.
+        # Take the contents of the src (could be file or directory)
+        # and concatinate it with the task config (or serialized_data).
+        #
+        # Pseudocode: hash(serialized_data + src_file)
+        #
+        # Note: serialized_data was grabbed above.
+        try:
+            task_config = json.loads(c)[task.action]
+            src_regex = "(?<=src\=).*?(?=\ )"
+            src = re.search(src_regex, task_config).group(0)
```

Comment:
it can also be dest or path

I'd love to see task (or task metadata) being accessed rather than going for regexes

Reply:
I'm grabbing the task metadata instead of regex. I hope this is an appropriate way to do so.
```
task_config = task.dump_attrs()
src = task_config['args']['src']
```

Reply:
Also, are you sure `dest` or `file` (which is only available for the [file module](https://docs.ansible.com/ansible/2.9/modules/file_module.html)) are appropriate? `dest` and `file` only interact with files on the container, but I only want to check for changes on files located on the host machine.